src/ResourceManager/ServiceBus/Commands.ServiceBus/Cmdlets/AuthorizationRule/GetAzureServiceBusAuthorizationRule.cs

```diff
@@ -0,0 +1,123 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ServiceBus.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ServiceBus.Commands
+{
+    /// <summary>
+    /// 'Get-AzureRmServiceBusAuthorizationRule' Cmdlet gives the details of a / List of AuthorizationRule(s)
+    /// <para> If AuthorizationRule name provided, a single AuthorizationRule detials will be returned</para>
+    /// <para> If AuthorizationRule name not provided, list of AuthorizationRules will be returned</para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ServiceBusAuthorizationRuleVerb, DefaultParameterSetName = NamespaceAuthoRuleParameterSet), OutputType(typeof(List<SharedAccessAuthorizationRuleAttributes>))]
+    public class GetAzureServiceBusAuthorizationRule : AzureServiceBusCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1, ParameterSetName = NamespaceAuthoRuleParameterSet,
+            HelpMessage = "Namespace Name.")]
+        [Parameter(ParameterSetName = QueueAuthoRuleParameterSet)]
+        [Parameter(ParameterSetName = TopicAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = QueueAuthoRuleParameterSet,
+            HelpMessage = "Queue Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasQueueName)]
+        public string Queue { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = TopicAuthoRuleParameterSet,
+            HelpMessage = "Topic Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasTopicName)]
+        public string Topic { get; set; }
+
+        [Parameter(Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "AuthorizationRule Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasAuthorizationRuleName)]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+
+            //Get Namespace Authorization Rule
+            if (ParameterSetName == NamespaceAuthoRuleParameterSet)
```

Comment:
@v-Ajnava it looks like this is still unresolved

```cs
if (ParameterSetName == NamespaceAuthoRuleParameterSet)
    if (!string.IsNullOrEmpty(Name))
    {
        ...
    }
    ...
```

instead of 

```cs
if (ParameterSetName == NamespaceAuthoRuleParameterSet)
{
    if (!string.IsNullOrEmpty(Name))
    {
        ...
    }
    ...
}
```